pennylane/circuit_drawer.py

```diff
@@ -0,0 +1,916 @@
+            layer (list): Layer that will be appended
+        """
+        self.raw_grid_transpose.append(layer)
+        self.raw_grid = _transpose(self.raw_grid_transpose)
+
+        return self
+
+    def replace_layer(self, idx, layer):
+        """Replace a layer in the Grid at the specified index.
+
+        Args:
+            idx (int): Index of the layer to be replaced
+            layer (list): Layer that replaces the old layer
+        """
+        self.raw_grid_transpose[idx] = layer
+        self.raw_grid = _transpose(self.raw_grid_transpose)
+
+        return self
+
+    def insert_wire(self, idx, wire):
+        """Insert a wire into the Grid at the specified index.
+
+        Args:
+            idx (int): Index at which to insert the new wire
+            wire (list): Wire that will be inserted
+        """
+        self.raw_grid.insert(idx, wire)
+        self.raw_grid_transpose = _transpose(self.raw_grid)
+
+        return self
+
+    def append_wire(self, wire):
+        """Append a wire to the Grid.
+
+        Args:
+            wire (list): Wire that will be appended
+        """
+        self.raw_grid.append(wire)
+        self.raw_grid_transpose = _transpose(self.raw_grid)
+
+        return self
+
+    @property
+    def num_layers(self):
+        """Number of layers in the Grid.
+
+        Returns:
+            int: Number of layers in the Grid
+        """
+        return len(self.raw_grid_transpose)
+
+    def layer(self, idx):
+        """Return the layer at the specified index.
+
+        Args:
+            idx (int): Index of the layer to be retrieved
+
+        Returns:
+            list: The layer at the specified index
+        """
+        return self.raw_grid_transpose[idx]
+
+    @property
+    def num_wires(self):
+        """Number of wires in the Grid.
+
+        Returns:
+            int: Number of wires in the Grid
+        """
+        return len(self.raw_grid)
+
+    def wire(self, idx):
+        """Return the wire at the specified index.
+
+        Args:
+            idx (int): Index of the wire to be retrieved
+
+        Returns:
+            list: The wire at the specified index
+        """
+        return self.raw_grid[idx]
+
+    def copy(self):
+        """Create a copy of the Grid.
+
+        Returns:
+            Grid: A copy of the Grid
+        """
+        return Grid(self.raw_grid.copy())
+
+    def append_grid_by_layers(self, other_grid):
+        """Append the layers of another Grid to this Grid.
+
+        Args:
+            other_grid (pennylane.circuit_drawer.Grid): Grid whos layers will be appended
+        """
+        for i in range(other_grid.num_layers):
+            self.append_layer(other_grid.layer(i))
+
+        return self
+
+    def __str__(self):
+        """String representation"""
+        ret = ""
+        for wire in self.raw_grid:
+            ret += str(wire)
+            ret += "\n"
+
+        return ret
+
+
+class CharSet(abc.ABC):
+    """Charset base class."""
```

Comment:
Here as the ``CharSet`` and its subclasses are easily separable from the rest of the logic, it might even be considered to organize them into a separate file (as currently there are 4-5 classes structured in the same file). Considering creating a ``pennylane/circuit_drawer`` directory and having these files there could also be an option.

Reply:
This is a good idea!

Reply:
I agree! subpackages are always preferable to a single module when they get to about this size